src/main/scala/com/spark3d/spatialOperator/SpatialQuery.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2018 Mayur Bhosale
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.spark3d.spatialOperator
+
+import com.spark3d.geometryObjects.Shape3D.Shape3D
+import com.spark3d.utils.GeometryObjectComparator
+import org.apache.spark.rdd.RDD
+
+import scala.collection.mutable.PriorityQueue
+
+
+object SpatialQuery {
+
+  def KNN[A <: Shape3D, B <:Shape3D](queryObject: A, rdd: RDD[B], k: Int): List[B] = {
+
+    val pq: PriorityQueue[B] = PriorityQueue.empty[B](new GeometryObjectComparator[B](queryObject.center))
+
+    val itr = rdd.toLocalIterator
+
+    while (itr.hasNext) {
+      val currentElement = itr.next
+      if (pq.size < k) {
+        pq.enqueue(currentElement)
+      } else {
+        val currentEleDist = currentElement.center.distanceTo(queryObject.center)
+        // TODO make use of pq.max
```

Comment:
on using pq.max, getting no implicit ordering defined for class B, even though we have defined one in GeometryObjectComparator. Any idea how we can resolve this???

Reply:
Hum, good question. Let me have a look at this today.

Reply:
I think I tried this, let me give this another shot. Will give an update here